sdk/servicebus/service-bus/src/clientEntityContext.ts

```diff
@@ -275,6 +275,7 @@ export namespace ClientEntityContext {
     };
 
     (entityContext as ClientEntityContext).close = async () => {
+      entityContext.isClosed = true;
```

Comment:
Just curious, why did you decide to move this to the top? Does this effectively prevent the user from calling operations on the client while it is closing?

Reply:
@HarshaNalluru Please update the PR description to capture all the nuances that led to the changes we are making in this PR

Reply:
Updated the description with relevant details. https://github.com/Azure/azure-sdk-for-js/pull/8537#issue-408782179
Let me know if that makes sense.

Reply:
Ah interesting, thanks for updating the description!